### ISSUE TITLE
[new feature] add command line options to bypass the data model selection dialog

### ISSUE THREAD

**Issue by jcaillon** [Wisser/Jailer#14]
Hello @Wisser ,

First of all, let me thank you and congratulate you for this amazing tool that is jailer. Really awesome work.

I would like to suggest a new feature: 

When you open Jailer or JailerDataBrowser, you get an initial dialog window that let you choose the data model/connection[/export model].
Would it be possible to bypass this dialog if the needed information are passed directly to the program through process arguments?

This would use roughly the same arguments as jailer.bat/.sh, something along those lines:

```
jailerDataBrowser.bat -jdbcjar db2jcc4.jar -driver=class.Driver -url="jdbc:xxx://localhost:xxx/database" -username=user -password=*** -datamodel "datamodel\xxx" -schemamapping SCHEMA_IN_DATAMODEL=SCHEMA_IN_DATABASE [-extractionmodel "extractionmodel\xxx.jm"]

jailerGUI.bat -jdbcjar db2jcc4.jar -driver=class.Driver -url="jdbc:xxx://localhost:xxx/database" -username=user -password=*** -datamodel "datamodel\xxx" -schemamapping SCHEMA_IN_DATAMODEL=SCHEMA_IN_DATABASE [-extractionmodel "extractionmodel\xxx.jm"]
```

What do you think about it?

Best regards

**Comment by Wisser** [comment]
Hi,

I also think that would be a good feature. Thanks for the suggestion.

However, at least the selection of the data model can already be skipped by selecting a connection in the initial dialog in the "Connection" tab. The data model associated with this connection is then automatically selected.
Similarly, an export model can be selected in the "Extraction Model" tab. The associated data model is then used.

**Comment by jcaillon** [comment]
Thank you for your answer, that's great to hear.

Even if you can simply select the connection and do not have to pick the correct data model from the dialog, you still need to manually input the connection settings.

In the use case I was thinking of, you would be able to configure several shortcuts that could instantly get you started in the databrowser/jailer with the right datamodel and the right connection. With no extra configuration needed at all.

To further explain my use case: I have an application with a database for which I have generated a data model. People working on this database install jailer locally in the same path (e.g. c:\jailer). On a shared drive, I put the datamodel for the database. I also create shortcuts that allow everyone to immediately connect to the dev/test/production databases through the databrowser. This way they never have to manually input the connection details and I can easily update those (and the datamodel) for everyone.

Best regards

**Comment by Wisser** [comment]
Thank you for the explanation. This is indeed a feature that would be useful to many users.
Btw, the connection information is saved in a file called ".connections". When you distribute this file, not every user needs to type in the connection details.

**Comment by jcaillon** [comment]
I was aware of the .connections file but it is not really handy to update it for everyone (as it is stored either in the jailer directory or the user directory with .multiusers if i'm correct).

It is also stored in binary format which requires it to be modified through the graphical interface.

It could be a fallback option however.

**Comment by jcaillon** [comment]
Hello @Wisser :)

Thank you very much for the implementation! I just tried it (v8.8.5) and found 2 issues:

---

### 1. data browser -schemamapping is not working

I'm trying the following:

```bash
jailerDataBrowser.bat -datamodel "C:\omega" -jdbcjar "C:\db2jcc4.jar" -driver com.ibm.db2.jcc.DB2Driver -url jdbc:db2://****:50000/OMB30ZD -user **** -password **** -schemamapping REFSCHEMA:EXISTINGSCHEMA
```

- The connection to the database is OK
- My data model a schema named **REFSCHEMA** (e.g. in table.csv i have REFSCHEMA.tablename)

If I try to open a table, the select fails because it is still trying to select from **REFSCHEMA** instead of **EXISTINGSCHEMA**.

If I change the schema name in all the data model files, it works as expected.

---

### 2. jailerGUI does not use the -datamodel correctly

I'm trying the following:

```bash
jailerGUI.bat "C:\omega\equipement.jm" -datamodel "C:\omega" -jdbcjar "C:\db2jcc4.jar" -driver com.ibm.db2.jcc.DB2Driver -url jdbc:db2://****:50000/OMB30ZD -user **** -password ****
```

Jailer does not start, I have the following exception:

```exception
java.lang.RuntimeException: Data model not found: file:/C:/jailer/omega/table.csv
	at net.sf.jailer.datamodel.DataModel.<init>(DataModel.java:392)
	at net.sf.jailer.datamodel.DataModel.<init>(DataModel.java:324)
	at net.sf.jailer.extractionmodel.ExtractionModel.<init>(ExtractionModel.java:200)
	at net.sf.jailer.extractionmodel.ExtractionModel.<init>(ExtractionModel.java:183)
	at net.sf.jailer.ui.ExtractionModelEditor.<init>(ExtractionModelEditor.java:270)
	at net.sf.jailer.ui.ExtractionModelFrame.<init>(ExtractionModelFrame.java:196)
	at net.sf.jailer.ui.ExtractionModelFrame.createFrame(ExtractionModelFrame.java:2105)
	at net.sf.jailer.ui.ExtractionModelFrame.createFrame(ExtractionModelFrame.java:2139)
	at net.sf.jailer.ui.ExtractionModelFrame$69.run(ExtractionModelFrame.java:2071)
```

It still checks if a datamodel "omega" exists, which is referenced in my extraction model (#! block datamodelfolder).

----

Tell me if I can help you to test it again.

best regards

**Comment by Wisser** [comment]
Hi @jcaillon 

**1. data browser -schemamapping is not working**

the syntax for the mapping is: "REFSCHEMA=EXISTINGSCHEMA,...".
There is a new tool in the menu "Tools" -> "Show Command Line", which should help to create a command line.

**2. jailerGUI does not use the -datamodel correctly**

That is indeed an error. If you like, feel free to test the correction. To do this, please replace the file "jailer.jar" with that from the zip archive in the attachment, or build the project from the repo.
[jailer.zip](https://github.com/Wisser/Jailer/files/3567973/jailer.zip)

**Comment by jcaillon** [comment]
**1. data browser -schemamapping is not working** 

-> totally my bad! Whoops. It works well with the right parameter.

**2. jailerGUI does not use the -datamodel correctly**

-> The correction works perfectly!

Thank you very much @Wisser 👍

**Comment by Wisser** [comment]
The feature has been released with version 8.8.6.
